### PR TITLE
Remove Firefox 47 saucelabs build

### PR DESCRIPTION
**Pull request by OndraM**
Exclusions and conditions in tests for the old FF were already removed in #863 , so now the old-FF tests on saucelabs are failing.

### CODE REVIEW

**Review by williamdes — APPROVED** [review]
We do not test anymore FF ?

**Comment by OndraM** [comment]
> We do not test anymore FF ?

We still do test against latest FF in integration tests. But this is removing Saucelabs build against the old FF 47. There is still OSS protocol build against saucelabs (via Chrome).

**Comment by williamdes** [comment]
> > We do not test anymore FF ?
> 
> We still do test against latest FF in integration tests. But this is removing Saucelabs build against the old FF 47. There is also still OSS protocol build against saucelabs (via Chrome).

But could you replace it with the Latest FF ?
Would that make sense ?

**Comment by andrewnicols** [comment]
Yeah, it would be good to test against latest Firefox. Maybe the tests can use a docker image instead of Saucelabs in future too? Would be faster than remote browser testing.

File: .github/workflows/sauce-labs.yaml [hunk 1]
```diff
@@ -15,13 +15,12 @@ jobs:
...
-          - { name: "Firefox 47, OSS protocol", BROWSER_NAME: "firefox", VERSION: "47.0", PLATFORM: "Windows 10", DISABLE_W3C_PROTOCOL: "1", tunnel-id: "gh-1" }
           # Chrome 74 is the last version which doesn't use W3C WebDriver by default and rather use OSS protocol
-          - { name: "Chrome 74, OSS protocol", BROWSER_NAME: "chrome", VERSION: "74.0", PLATFORM: "Windows 10", DISABLE_W3C_PROTOCOL: "1", tunnel-id: "gh-2" }
-          - { name: "Chrome latest, W3C protocol", BROWSER_NAME: "chrome", VERSION: "latest", PLATFORM: "Windows 10", tunnel-id: "gh-3" }
-          - { name: "Edge latest, W3C protocol", BROWSER_NAME: "MicrosoftEdge", VERSION: "latest", PLATFORM: "Windows 10", tunnel-id: "gh-4" }
+          - { name: "Chrome 74, OSS protocol", BROWSER_NAME: "chrome", VERSION: "74.0", PLATFORM: "Windows 10", w3c: false, tunnel-id: "gh-1" }
+          - { name: "Chrome latest, W3C protocol", BROWSER_NAME: "chrome", VERSION: "latest", PLATFORM: "Windows 10", tunnel-id: "gh-2" }
+          - { name: "Edge latest, W3C protocol", BROWSER_NAME: "MicrosoftEdge", VERSION: "latest", PLATFORM: "Windows 10", tunnel-id: "gh-3" }
```
Comment (andrewnicols) [inline .github/workflows/sauce-labs.yaml:23]: Would be good t have a Firefox latest version here.

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -108,7 +107,6 @@ jobs:
           php-version: 7.4
           extensions: mbstring, intl, zip
           coverage: xdebug
-          tools: composer:v2
```
Comment (andrewnicols) [inline .github/workflows/tests.yaml:111]: Is this now the default or is this an unrelated change?

**Review by andrewnicols — CHANGES_REQUESTED** [review]

File: .github/workflows/tests.yaml (hunk 2 above)
Comment (williamdes) [reply .github/workflows/tests.yaml:111]: Maybe a copy paste from my example :smile: 
I forced it on phpMyAdmin to have more performance and have less time spent installing the composer modules

Comment (andrewnicols) [reply .github/workflows/tests.yaml:111]: But that's a removal of v2 - so slower.

**Comment by OndraM** [comment]
> Yeah, it would be good to test against latest Firefox. Maybe the tests can use a docker image instead of Saucelabs in future too? Would be faster than remote browser testing.

But we still test against latest firefox, there are two build in functional tests:
- https://github.com/php-webdriver/php-webdriver/runs/1910337096?check_suite_focus=true (via Selenium server)
- https://github.com/php-webdriver/php-webdriver/runs/1910337115?check_suite_focus=true (only with geckodriver)

This is latest Firefox is installed directly in Ubuntu VM used by GH Actions, so this is probalby even faster then in docker and we are closer to real Firefox deployment than Firefox hidden in Saucelabs.

The change here is we don't test against Firefox via Saucelabs. I don't see there real value: Saucelabs is mainy for browsers/versions, we cannot run locally (which is always faster and less flaky). This is now mainly MS Edge on Windows. The two other saucelab builds are Chrome on Windows with OSS and Chrome on Windows with W3C protocol, but I take them as a sanity check - to make sure we don't break anything for people running tests against Saucelabs (as they have their custom implementation of the remote selenium endpoint, with different behavior than Selenium server). 

I would be really considerate with any other saucelabs build - they are much slower and less reliable. And especially if we can achieve the same locally or in docker, it should be preferred. And we already have latest Firefox build. What would be a value of such additional build with the same browser, would it be worth the cost and effort?

File: .github/workflows/tests.yaml (hunk 2 above)
Comment (OndraM) [reply .github/workflows/tests.yaml:111]: >  The latest stable version of composer is set up by default.
https://github.com/marketplace/actions/setup-php-action#wrench-tools-support

And Composer 2 is in the stable channel for a while. See build output:
https://github.com/php-webdriver/php-webdriver/runs/1910336904?check_suite_focus=true#step:3:12

**Comment by andrewnicols** [comment]
In which case +1 :)

**Review by andrewnicols — APPROVED** [review]

**Review by williamdes — APPROVED** [review]

**Comment by andrewnicols** [comment]
@OndraM , do you need any more from us?

**Comment by OndraM** [comment]
> @OndraM , do you need any more from us?

No, only a time :smile: . Merging now :rocket:
